### DISCUSSION TITLE
Hytera RRS answer message to a radio request, format details

### DISCUSSION THREAD

**Discussion by cmolax1090** [OK-DMR/ok-dmrlib#10] [Q&A] [answered]
Hi,

I would like to be able to reply to a Hytera RRS request sent by a radio terminal to a UDP server.
By studying the code of the dmrlib library I understood how the RRS request is formatted and thanks to Wireshark I discovered what must be answered to the radio by the server.

In practice: 
The radio announces itself with a message like this:
[1100030004180001937f03]
The library's rrs utility allows me to understand the request:

dmrlib-hytera-rrs 1100030004180001937f03
1100030004180001937f03
[RRSTypes.RadioRegistrationRequest] [RadioIP subnet:24 id:403]

Ideally, you should reply with a Timeout value, after which the radio should announce itself again.

The response can look like this message, in this example we ask it to announce itself again in 900 seconds (x384 seconds):
[11008000091800019300000003847603]

dmrlib-hytera-rrs 11008000091800019300000003847603
11008000091800019300000003847603
[RRSTypes.RadioRegistrationAnswer] [RadioIP subnet:24 id:403] [RESULT: RRSResult.Success] [RENEW PERIOD: 900s]

This response message includes the radio's address and the 900-second confirmation. The radio will send a new announcement message in 900 seconds.

If we don't reply correctly to the radio's RRS request, it will send a request every 20 seconds for a certain period of time.

I've managed to figure out how to format the response to a radio's RRS request, but I'm having trouble with part of the message. In my example above, I can't figure out how to fill in the 15th byte in this case [76] of this response, iIf it's a checksum, how is it calculated ?
.
The "dmrlib-hytera-rrs" utility doesn't seem to care about this byte 15 and ignores it.

Where can I find this information?

Thank you in advance for your valuable advice.

**Comment by smarek** [comment, marked as answer]
Hi @cmolax1090 

it's quite simple, RRS is HDAP, https://github.com/OK-DMR/ok-dmrlib/blob/master/okdmr/dmrlib/hytera/pdu/hdap.py#L63

However, why don't you just construct the payload using ok-dmrlib? Like this https://github.com/OK-DMR/ok-dmrlib/blob/master/okdmr/tests/dmrlib/hytera/pdu/test_rrs.py#L28

**Reply by cmolax1090** [reply]
Hi smarek,

Thank you so much for your valuable advice, which has helped me
solve my problem.

**Reply by smarek** [reply]
@cmolax1090 you're welcome, if your adventure yielded something, you could contribute back to the opensource, please do. Any kind of working scripts, more example payloads for any protocol we have or might have in the future.
